python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -562,32 +563,34 @@ def get_asset_partitions_updated_after_cursor(
         previous_versions = self._asset_partitions_data_versions(
             asset_key, updated_after_cursor, before_cursor=after_cursor + 1
         )
-        return [
+        return {
             asset_partition
             for asset_partition, version in latest_versions.items()
             if previous_versions.get(asset_partition) != version
-        ]
+        }
 
     @cached_method
-    def is_reconciled(self, *, asset_partition: AssetKeyPartitionKey) -> bool:
-        """Returns a boolean representing if the given `asset_partition` is currently reconciled.
-        An asset (partition) is considered unreconciled if any of:
-        - It has never been materialized
-        - One of its parents has been updated more recently than it has
-        - One of its parents is unreconciled.
+    def to_be_materialized_before_reconciled(
```

Comment:
Would it make sense to call this `get_unreconciled_ancestors`?

Reply:
went w/ get_unreconciled_root_ancestors